File: Binary Search/binary_search.dart
void main() {
  int target = 100;
  List<int> arr = [-1, 0, 4, 6, 11, 33, 53, 55];
  int leftP = 0;
  int rightP = arr.length - 1;
  search() {
    while (leftP <= rightP) {
      var mid = ((leftP + rightP) / 2).round();
      if (arr[mid] == target) {
        return mid;
      }
      if (target < arr[mid]) {
        rightP = mid - 1;
      } else {
        leftP = mid + 1;
      }
    }
    return -1; //not found target
  }

  print(search());
}

File: Binary Search/binarySearch.js
/**
 * Binary Search in JavaScript
 * @param {*} arr array of numbers sorted in ascending order
 * @param {*} searchValue number to search
 * @return {*} index of the search value or -1 if not found
 */
const binarySearch = function (arr, searchValue) {
    let min = 0;
    let max = arr.length - 1;
    let mid;
    while (min <= max) {
        mid = (min + max) >>> 1;
        if (arr[mid] === searchValue) {
            return mid;
        } else if (arr[mid] < searchValue) {
            min = mid + 1;
        } else {
            max = mid - 1;
        }
    }
    return -1;
};
